### PR TITLE
chore: use Express serving static files as single page application

### PR DESCRIPTION
**Pull request by tyyzqmf**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary

(describe what this merge request does)

## Implementation highlights

(describe how the merge request does for feature changes, share the RFC link if it has)

## Test checklist

- [ ] add new test cases
- [ ] all code changes are covered by unit tests
- [x] end-to-end tests
  - [ ] deploy web console with CloudFront + S3 + API gateway
  - [x] deploy web console within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [ ] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [ ] deploy with reporting
  - [ ] streaming ingestion
    - [ ] with Redshift Serverless
    - [ ] with provisioned Redshift

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

File: frontend/serve/package.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{
+  "name": "frontend-serve",
```
Comment (zxkane) [inline frontend/serve/package.json:2]: manage it via projen too?

File: src/control-plane/alb-lambda-portal.ts [hunk 2]
```diff
@@ -376,21 +376,23 @@ export class ApplicationLoadBalancerLambdaPortal extends Construct {
       ],
     );
 
-    const lambdaFn = new DockerImageFunction(this, 'portal_fn', {
+    const lambdaFn = new LambdaFunction(this, 'portal_fn', {
```
Comment (zxkane) [inline src/control-plane/alb-lambda-portal.ts:379]: Could we use `SolutionNodejsFunction` to share the configuration of Lambda?

File: src/control-plane/frontend/Dockerfile [hunk 3]
```diff
@@ -1,7 +1,4 @@
-ARG ADAPTER_VERSION=0.8.2
-FROM --platform=linux/aarch64 public.ecr.aws/awsguru/aws-lambda-adapter:$ADAPTER_VERSION-aarch64 AS adapter
-
-FROM public.ecr.aws/docker/library/node:20 AS builder
+FROM public.ecr.aws/docker/library/node:20
```
Comment (zxkane) [inline src/control-plane/frontend/Dockerfile:1]: Should we keep it as builder to save the runtime layer size?

Comment (tyyzqmf) [reply src/control-plane/frontend/Dockerfile:1]: This Dockerfile use at build stage, not deploy to lambda.

File: src/control-plane/alb-lambda-portal.ts (hunk 2 above)
Comment (tyyzqmf) [reply src/control-plane/alb-lambda-portal.ts:379]: SolutionNodejsFunction not support `code: this.getLambdaCode()`

File: frontend/serve/package.json (hunk 1 above)
Comment (tyyzqmf) [reply frontend/serve/package.json]: done

**Comment by zxkane** [comment]
### SonarQube Quality Gate Result
- **Result**: :exclamation: Error
- Triggered by @tyyzqmf on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|Reliability rating|:white_check_mark: OK|1|> 1|
|Security rating|:white_check_mark: OK|1|> 1|
|Sqale rating|:white_check_mark: OK|1|> 1|
|Coverage|:exclamation: Error|73.50|< 80|
|Duplicated lines density|:white_check_mark: OK|5.20|> 30|
|Blocker violations|:white_check_mark: OK|0|> 0|
|Bugs|:white_check_mark: OK|0|> 0|
|Code smells|:white_check_mark: OK|8|> 40|
|Critical violations|:white_check_mark: OK|0|> 0|
|Major violations|:white_check_mark: OK|0|> 0|
|Vulnerabilities|:white_check_mark: OK|0|> 0|

[View on SonarQube](http://sonarqube:9000/dashboard?id=pr-1417)
###### _updated: 6/6/2024, 07:56:59 (UTC+0)_

**Comment by zxkane** [comment]
# SonarQube Code Analytics
## Quality Gate failed

failed

## Additional information
*The following metrics might not affect the Quality Gate status but improving them will improve your project code quality.*

## Issues
bug   [0 Bugs](http://sonarqube:9000/project/issues?id=pr-1417&resolved=false&sinceLeakPeriod=true&types=BUG)

vulnerability   [0 Vulnerabilities](http://sonarqube:9000/project/issues?id=pr-1417&resolved=false&sinceLeakPeriod=true&types=VULNERABILITY)

code_smell   [0 Code Smells](http://sonarqube:9000/project/issues?id=pr-1417&resolved=false&sinceLeakPeriod=true&types=CODE_SMELL)

## Coverage and Duplications
*No data* **Coverage**

*No data* **Duplication**

**Comment by zxkane** [comment]
### SonarQube Quality Gate Result
- **Result**: :exclamation: Error
- Triggered by @tyyzqmf on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|Reliability rating|:white_check_mark: OK|1|> 1|
|Security rating|:white_check_mark: OK|1|> 1|
|Sqale rating|:white_check_mark: OK|1|> 1|
|Coverage|:exclamation: Error|73.50|< 80|
|Duplicated lines density|:white_check_mark: OK|5.20|> 30|
|Blocker violations|:white_check_mark: OK|0|> 0|
|Bugs|:white_check_mark: OK|0|> 0|
|Code smells|:white_check_mark: OK|8|> 40|
|Critical violations|:white_check_mark: OK|0|> 0|
|Major violations|:white_check_mark: OK|0|> 0|
|Vulnerabilities|:white_check_mark: OK|0|> 0|

[View on SonarQube](http://sonarqube:9000/dashboard?id=pr-1417)
###### _updated: 6/6/2024, 08:58:35 (UTC+0)_

File: frontend-serve/README.md [hunk 4]
```diff
@@ -0,0 +1 @@
+# replace this
```
Comment (zxkane) [inline frontend-serve/README.md:1]: update it

**Review by zxkane — CHANGES_REQUESTED** [review]

**Review by zxkane — APPROVED** [review]

**Pull request merged**